packages/bento-design-system/src/util/LocalizedString.ts

```diff
@@ -1,9 +1,12 @@
-/**
- * A string that has been localized
- */
-export type LocalizedString = string & {
-  readonly LocalizedString: "LocalizedString";
-};
+interface ConfigurableTypes {
+  LocalizedString: string;
+}
+
+export interface TypeOverrides {}
+
+type ResolvedTypes = Omit<ConfigurableTypes, keyof TypeOverrides> & TypeOverrides;
```

Comment:
this is a trick I borrowed from react-i18next: we compute the final type as the merge of two objects. The second one (`TypeOverrides`) takes precedence and it’s initially empty.

Users of the library can then use module augmentation to inject their own types and they will “win” over the default one.